xclim/testing/helpers.py

```diff
@@ -10,13 +12,33 @@
 from pathlib import Path
 from shutil import copytree
 from sys import platform
+from typing import Any
+from urllib.error import HTTPError, URLError
+from urllib.parse import urljoin, urlparse
+from urllib.request import urlretrieve
 
 import numpy as np
 import pandas as pd
 import xarray as xr
-from dask.diagnostics import Callback
+from dask.callbacks import Callback
 from filelock import FileLock
 from packaging.version import Version
+from xarray import Dataset
+from xarray import open_dataset as _open_dataset
+
+try:
+    from pytest_socket import SocketBlockedError
+except ImportError:
+    SocketBlockedError = None
+
+try:
+    import pooch
+except ImportError:
+    warnings.warn(
+        "The `pooch` library is not installed. "
+        "The default cache directory for testing data will not be set."
+    )
+    pooch = None
```

Comment:
what happens then? not cache, we always download?

Reply:
If pooch isn't providing a path, then a user can hardcode one and copy the data to a fixed location manually. If no data is found, then it (rightfully) will crash with an OSError.